scene/resources/syntax_highlighter.cpp

```diff
@@ -652,3 +653,120 @@ void CodeHighlighter::set_member_variable_color(Color p_color) {
 Color CodeHighlighter::get_member_variable_color() const {
 	return member_color;
 }
+
+////////////////////////////////////////////////////////////////////////////////
+
+Dictionary GDShaderSyntaxHighlighter::_get_line_syntax_highlighting_impl(int p_line) {
+	Dictionary color_map;
+
+	const String &str = text_edit->get_line(p_line);
+	const int line_length = str.length();
+
+	for (const BranchRegion &region : disabled_branch_regions) {
+		if (p_line >= region.from_line && p_line <= region.to_line) {
+			Dictionary highlighter_info;
+			highlighter_info["color"] = disabled_branch_color;
+
+			for (int i = 0; i < line_length; i++) {
+				color_map[i] = highlighter_info;
+			}
```

Comment:
There shouldn't be a need to loop here as the highlighter will only stop using the current colour when it reaches a new line, or a new entry in the `color_map`. Given we're setting the entire line the same colour, `color_map[0] = highlighter_info;` should work.